### PR TITLE
define getName() for UI subclasses and add opt debug to matrixdriver

### PR DESCRIPTION
**Pull request by mnantel**
definitions_cxx.hpp -> #define MATRIX_DRIVER_DEBUG enables additional debug output
names are the file name of the class, usually the same as the class name and the UI mode name, but lowercase
debug prints out getName from UI subclass, pad x/y, velocity

the purpose of this change is to facilitate learning the code workflow

### CODE REVIEW

File: src/deluge/gui/views/instrument_clip_view.cpp [hunk 1]
```diff
@@ -3470,6 +3470,11 @@ void InstrumentClipView::auditionPadAction(int32_t velocity, int32_t yDisplay, b
 		actionLogger.closeAction(ActionType::NOTE_NUDGE);
 	}
 
+	if (shiftButtonDown && velocity) {
+		D_PRINT("AUTOSAMPLE", velocity);
+		display->displayPopup("AUTOSAMPLE");
+		return;
+	}
```
Comment (sapphire-arches) [inline src/deluge/gui/views/instrument_clip_view.cpp:3477]: Leftover from some earlier change?

File: src/definitions_cxx.hpp [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
 #define AUTOMATED_TESTER_ENABLED (0 && ALPHA_OR_BETA_VERSION)
 
+#define MATRIX_DRIVER_DEBUG 0
+
```
Comment (sapphire-arches) [inline src/definitions_cxx.hpp:31]: This could be plumbed in to the build system - look for how `ENABLE_SYSEX_LOAD` works for an example.

Comment (mnantel) [reply src/definitions_cxx.hpp:31]: Removed the gremlin. I can see about the build system element.

**Comment by mnantel** [comment]
Ok, I think I am good with this PR. Lesson learnt: keep those as drafts until I am done :) Sorry about the spam and numerous commits. Live and learn!

**Comment by m-m-adams** [comment]
This looks good to me. It takes up about 800 bytes for the extra strings but I think removing unused strings in release builds is probably a more general optimization we should look into

**Review by m-m-adams — APPROVED** [review]

**Pull request merged**